### PR TITLE
[Experimental] Apache Arrow for Numpy arrays

### PR DESCRIPTION
**Pull request by tudorcebere**
## Description
Please include a summary of the change, the motivation, and any additional context that will help others understand your PR. If it closes one or more open issues, [please tag them as described here](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

**Comment by gmuraru** [comment]
Looking forward to this! This means that if we want to send a tensor, it would make more sense to convert it to numpy --> use arrow --> convert back to tensor?

**Comment by tudorcebere** [comment]
from torch.utils import benchmark
import numpy as np
from syft import serialize, deserialize, load
from syft.experimental_flags import flags
load("numpy")


def serde(obj):
    bytes_repr = serialize(obj, to_bytes=True)
    obj_back = deserialize(bytes_repr, from_bytes=True)


results = []

for backend in ["Arrow", "Proto"]:
    if backend == "Arrow":
        flags.APACHE_ARROW_TENSOR_SERDE = True
    else:
        flags.APACHE_ARROW_TENSOR_SERDE = False

    for dtype in [np.float16, np.float32, np.float64]:
        for sizes in [(100, 100), (500, 500), (1000, 1000), (100, 100, 100), (50, 50, 50, 50)]:
            obj = np.random.randn(*sizes).astype(dtype)
            t = benchmark.Timer(
                stmt='serde(obj)',
                setup='from __main__ import serde',
                label=f'Serde of Numpy arrays',
                globals={'obj': obj},
                sub_label=f'dtype: {dtype.__name__}, shape: {sizes}',
                description=backend
            ).timeit(500)
            results.append(t)
            print(f"Finished {backend} - {dtype} - {sizes} on 1 thread")

compare = benchmark.Compare(results)
compare.print()

**Comment by tudorcebere** [comment]
@gmuraru if you are sending it using Syft, we'll add an experimental module for this. There are a few caveats on this:
* they have support only for cpp and python on it (js/java is on the roadmap, but they are not ready)

On the other side, they have amazing support for:
* compression
* zero-copy operations
* pandas
* GPU

**Review by madhavajay — APPROVED** [review]
nice! Experimental flags for the win.
